lib/driver.js

```diff
@@ -642,6 +642,15 @@ class XCUITestDriver extends BaseDriver {
                `This may cause problems if a simulator does not exist for this platform version.`);
       this.opts.platformVersion = this.iosSdkVersion;
     }
+
+    if (this.opts.noReset) {
```

Comment:
i'm curious why this change is restricted to the case where `noReset` is true. why wouldn't we want to use an existing sim with the default platformVersion in _any_ case we can?

Reply:
This is because I'm assuming that in cases where noReset is false you do not want to just use the default simulators matchinging the capabilities, and would rather have an isolated simulator.

If this isn't the case, we can actually I think make this even simpler by moving lines 649-644 where it gets the capabilities up to above the first "getExistingSim."  I did that as well and it works, but I was afraid it might break existing functionality.

Reply:
Yes, that's a good thought to avoid changing the existing behavior. Hmm, what do you think @imurchie?

Reply:
Any further thoughts on this? I can modify if needed.

Reply:
That sounds fine to me.